### DISCUSSION TITLE
How to handle `elastalert_status_status` index growing too large?

### DISCUSSION THREAD

**Discussion by archfz** [jertel/elastalert2#713] [Q&A] [answered]
My question would be can this index be rolled over and older versions deleted ?

**Comment by archfz** [comment, marked as answer]
My issue is actually that I do not see an index template for this index. And thus I don't know how to attach an ILM policy to it persistently.

**Reply by nsano-rururu** [reply]
>My issue is actually that I do not see an index template for this index. And thus I don't know how to attach an ILM policy to it persistently.

Not Support

**Reply by nsano-rururu** [reply]
Currently, I think there is no choice but to delete the elastalert index and re-execute the elastalert to create the elastalert index.

**Reply by archfz** [reply]
So I guess this could be a feature request?

**Reply by nsano-rururu** [reply]
There are no plans for response. Pull requests are welcome.

**Reply by nsano-rururu** [reply]
It seems that they were trying to implement something like ILM Policy on their own with the original yelp / elastalert, but it seems that development was interrupted in the middle.

**Reply by lymanaukuleles** [reply]
You could possibly put the Kibana Dev Tool Delete Index API into a script on your server and have it delete the index (or documents in the index) periodically. Actually you cannot delete the Elastalert indices or it will crash right? So it would have to delete only the documents if that is possible.

**Reply by nsano-rururu** [reply]
[Added elastalert_status index timestamping #945](https://github.com/Yelp/elastalert/pull/945)
[Is it possible to have daily "elastalert_status-%{+YYYY.MM.dd}" indices #684](https://github.com/Yelp/elastalert/issues/684)
[Size of ElastAlert Metadata Index #178](https://github.com/Yelp/elastalert/issues/178)

**Comment by aeftef** [comment]
In my installation, I am running a cron that deletes elastalert documents older than x ammount of days. (using a delete_by_query... yeaa...  its not the bast practice, but good enough)

As far as I know it is not causing any problem.

Take into account that the time range my alerts are valid is not overlapping with the delete time range (my alerts makes sense at most in the range of a couple of days, and im deleting stuff weeks old).

So maybe you can do something similar, adjusting the time range where your alerts make sense

**Comment by iamxeph** [comment]
Could we just switch to [Data Streams](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html)? I'm using it for all other indice at work.
